lib/matplotlib/backends/backend_wx.py

```diff
@@ -1374,29 +1374,28 @@ def save_figure(self, *args):
         # Fetch the required filename and file type.
         filetypes, exts, filter_index = self.canvas._get_imagesave_wildcards()
         default_file = self.canvas.get_default_filename()
-        dlg = wx.FileDialog(self.canvas.GetParent(),
-                            "Save to file", "", default_file, filetypes,
-                            wx.FD_SAVE | wx.FD_OVERWRITE_PROMPT)
+        dlg = wx.FileDialog(
+            self.canvas.GetParent(), "Save to file",
+            rcParams["savefig.directory"], default_file, filetypes,
+            wx.FD_SAVE | wx.FD_OVERWRITE_PROMPT)
         dlg.SetFilterIndex(filter_index)
         if dlg.ShowModal() == wx.ID_OK:
-            dirname = dlg.GetDirectory()
-            filename = dlg.GetFilename()
-            _log.debug('%s - Save file dir:%s name:%s',
-                       self, dirname, filename)
-            format = exts[dlg.GetFilterIndex()]
-            basename, ext = os.path.splitext(filename)
-            if ext.startswith('.'):
-                ext = ext[1:]
-            if ext in ('svg', 'pdf', 'ps', 'eps', 'png') and format != ext:
+            path = pathlib.Path(dlg.GetPath())
+            _log.debug('%s - Save file path: %s', type(self), path)
+            fmt = exts[dlg.GetFilterIndex()]
+            ext = path.suffix[1:]
+            if ext in self.canvas.get_supported_filetypes() and fmt != ext:
                 # looks like they forgot to set the image type drop
                 # down, going with the extension.
                 _log.warning('extension %s did not match the selected '
                              'image type %s; going with %s',
-                             ext, format, ext)
-                format = ext
+                             ext, fmt, ext)
+                fmt = ext
+            # Save dir for next time, unless empty str (which means use cwd).
+            if rcParams["savefig.directory"]:
```

Comment:
You seem to have a different philosophy abut `rcParams` than I do.  I think of it as defaults the user/matplotlib have set, not something we change as the GUI gets run.  i.e. if I set `save fig.directory` in my matplotlibrc, I don't expect it to change?

Reply:
That has been the way in all backends. From a users point of view, remembering the last used directory in a save dialog is the expected (=standard) behavior for an application.

One can argue if that's should still be the case between multiple runs of the program. But that would be a change with respect to the current behavior.

Reply:
Oh duh, sorry didn't notice that it said the other backends do that.  Sorry.